site/docs/master/install-overview.md

```diff
@@ -39,6 +41,10 @@ To see the YAML applied by the `velero install` command, use the `--dry-run -o y
 
 For more complex installation needs, use either the generated YAML, or the Helm chart.
 
+### Installing object storage or volume snapshotter plugins
+
+`velero install` _will not_ install any object storage or volume snapshotter plugins. In order to install these, you must run `velero plugin add <URL of container here>` after running `velero install`.
+
```

Comment:
I would say something like: "As of v1.12", to emphasize that this is now new.

I would not be against adding the `plugins` flag to the `install` cmd. That would make the interaction more user friendly since if it wasn't included we would warn immediately that it needs to be added. 

To note: if we don't add this flag, then logging would need to be added to warn the user during creation of backups in case the `plugin add` was not run.

Reply:
What would we log? I can run some experiments, but it should error because the plugin won't be found for either the object storage or volume snapshotter.

Reply:
Sorry, yes, I meant error out, to warn the user. I'm not crazy about this flow, in other words, telling the user there was an error w/o first preventing them from triggering the error. Some ppl don't read the docs.

Reply:
`velero install --bucket $BUCKET --provider velero.io/invalid --secret-file bogus_file --wait` will error out due to the behavior that #1883 seeks to address

```
An error occurred: some backup storage locations are invalid: error getting backup store for location "default": unable to locate ObjectStore plugin named velero.io/invalid
```

So the pod will crashloop until the ObjectStore plugin is added.

I'm not finding any validation for VSLs, so I don't think there would be an error thrown for finding a VolumeSnapshotter plugin until a backup is taken that includes a volume.

Reply:
Ok. Would it be worthwhile adding to that error msg something like "please make sure you run `plugin add` to add a plugin for your provider"?

Reply:
Yeah, that seems reasonable to me, even if we add the `--plugins` flag. People could forget to add that at install time, and need to add it after the fact, so I think providing a hint when we can't find a plugin is good no matter what.

Reply:
If we had the `--plugins` flag, would it not be required? I'd think it should.

Reply:
True, it should be required. Still, I don't think adding the hint on missing plugins hurts. Something like this:

```diff
diff --git i/pkg/plugin/clientmgmt/registry.go w/pkg/plugin/clientmgmt/registry.go
index 620d08ad..bdf506f0 100644
--- i/pkg/plugin/clientmgmt/registry.go
+++ w/pkg/plugin/clientmgmt/registry.go
@@ -228,7 +228,7 @@ func newPluginNotFoundError(kind framework.PluginKind, name string) *pluginNotFo
 }
 
 func (e *pluginNotFoundError) Error() string {
-	return fmt.Sprintf("unable to locate %v plugin named %s", e.kind, e.name)
+	return fmt.Sprintf("unable to locate %v plugin named %s. Use `velero plugin add <IMAGE_LOCATION>` to add a plugin", e.kind, e.name)
 }
 
 type duplicatePluginRegistrationError struct {
```

Reply:
I thought we didn't like the idea of the `--plugins` flag because it was now a little confusing whether that related to the BSL or the VSL or both?

Reply:
Ah yes. These tradeoffs are a nightmare. Today I was also thinking about this new issue/PR: https://github.com/heptio/velero/pull/1883 and maybe what we should do is NOT add the ``plugins` flag, document, and let the backup validation fail if the plugins wasn't added.

Reply:
Let's move comments on `velero install --plugins` to #1740 - see https://github.com/heptio/velero/issues/1740#issuecomment-533180281